internal/consul/lock_test.go

```diff
@@ -0,0 +1,146 @@
+package consul
+
+import (
+	"testing"
+
+	consulApi "github.com/hashicorp/consul/api"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+var fakeLockGetAttemps int = 0
+var globalT *testing.T
+
+func increaseAttempts() {
+	fakeLockGetAttemps++
+}
+
+func fakeLockGet(c *client, prefix string, q *consulApi.QueryOptions) (*consulApi.KVPair, *consulApi.QueryMeta, error) {
+	defer increaseAttempts()
+
+	assert.Equal(globalT, prefix, "test")
```

Comment:
I'm not clear: why are we asserting stuff inside this mock function? Also, couldn't we pass `t` as an argument to a function returning the fake function, instead of using global? Having a shared `*testing.T` is a dangerous ticking bomb!
I think we could very well avoid this assertion entirely.

Reply:
I cannot pass the variable as argument, this would change the faked function signature, which we cannot.
I could remove the `assert` but there is not any other place to do it. I didn't want to create a full mock, which has this functionalities